### PR TITLE
Decaffeinate files for plugins async error handling improvements

### PR DESCRIPTION
**Pull request by chrisbreiding**
This decaffeinates files in preparation for the work for #6570

# DO NOT SQUASH

### CODE REVIEW

File: packages/server/lib/controllers/spec.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      }
+
+      const filePath = err.filePath != null ? err.filePath : spec
+
+      err = errors.get('BUNDLE_ERROR', filePath, preprocessor.errorMessage(err))
+
+      console.log('') // eslint-disable-line no-console
+      errors.log(err)
+
+      return project.emit('exitEarlyWithErr', err.message)
```
Comment (flotwig) [inline packages/server/lib/controllers/spec.js]:
```suggestion
      project.emit('exitEarlyWithErr', err.message)
```

File: packages/server/lib/routes.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+const staticCtrl = require('./controllers/static')
+
+module.exports = function (app, config, request, getRemoteState, getDeferredResponse, project, networkProxy) {
+  // routing for the actual specs which are processed automatically
+  // this could be just a regular .js file or a .coffee file
+  app.get('/__cypress/tests', (req, res, next) => {
+    // slice out the cache buster
+    const test = CacheBuster.strip(req.query.p)
+
+    return spec.handle(test, req, res, config, next, project)
```
Comment (flotwig) [inline packages/server/lib/routes.js]: not 100% sure that this controller/other controllers need to be `return`ed, do we use some express middleware that makes returning a promise from a controller meaningful?

**Comment by Bkucera** [comment]
squashed into 3 commits instead of 6

File: packages/server/lib/routes.js (hunk 2 above)
Comment (chrisbreiding) [reply packages/server/lib/routes.js]: Probably not. I'll remove them. Any issues should be caught be tests.

**Pull request merged**